ddtrace/span.py

```diff
@@ -62,6 +63,7 @@ class Span(object):
         "_context",
         "_parent",
         "_ignored_exceptions",
+        "_callbacks",
```

Comment:
I'm confused by this `_callbacks` naming `on_span_finish` which seems more clear. I agree that we should move this logic into the `Span.finish()` but would prefer it be specified as a `on_finish` hook.

Reply:
sure, i suppose there could be other kinds of callbacks in the future :+1:

Reply:
I've moved this change out to #2198